nimare/transforms.py

```diff
@@ -179,10 +184,128 @@ def resolve_transforms(target, available_data, masker):
             return None
         varcope = masker.inverse_transform(varcope.squeeze())
         return varcope
+    elif target == "p":
+        if ("t" in available_data.keys()) and ("sample_sizes" in available_data.keys()):
+            dof = sample_sizes_to_dof(available_data["sample_sizes"])
+            t = masker.transform(available_data["t"])
+            z = t_to_z(t, dof)
+            p = z_to_p(z)
+        elif "z" in available_data.keys():
+            z = masker.transform(available_data["z"])
+            p = z_to_p(z)
+        else:
+            return None
+        p = masker.inverse_transform(p.squeeze())
+        return p
     else:
         return None
 
 
+class CoordinateGenerator(Transformer):
+    """Extract peak statistical coordinates from statistical z or p maps.
+
+    Parameters
+    ----------
+    cluster_threshold : :obj:`int` or `None`, optional
+        Cluster size threshold, in voxels. Default=None
+    min_distance : :obj:`float`, optional
+        Minimum distance between subpeaks in mm. Default=8mm.
+    overwrite : :obj:`bool`
+        If true, overwrites existing coordinate reports for study contrasts
+        with statistical images. Default=False.
+    z_threshold : :obj:`float`
+        Cluster forming z-scale threshold. Default=3.1.
+
+    Returns
+    -------
+    coordinates_df : :class:`pandas.DataFrame`
+        DataFrame containing statistical peaks.
+    """
+
+    def __init__(self, cluster_threshold=None, min_distance=8.0, overwrite=False, z_threshold=3.1):
+        self.cluster_threshold = cluster_threshold
+        self.min_distance = min_distance
+        self.overwrite = overwrite
+        self.z_threshold = z_threshold
+
+    def transform(self, dataset):
+        """Create coordinate peaks from statistical images."""
+        # relevant variables from dataset
+        space = dataset.space
+        masker = dataset.masker
+        images_df = dataset.images
+
+        # conform space specification
+        if "mni" in space.lower() or "ale" in space.lower():
+            coordinate_space = "MNI"
+        elif "tal" in space.lower():
+            coordinate_space = "TAL"
+        else:
+            coordinate_space = None
+
+        coordinates_dict = {}
+        for _, row in images_df.iterrows():
+
+            if row["id"] in list(dataset.coordinates["id"]) and not self.overwrite:
+                continue
+
+            if row.get("z"):
+                clusters = get_clusters_table(
+                    nib.funcs.squeeze_image(nib.load(row.get("z"))),
+                    self.z_threshold,
+                    self.cluster_threshold,
+                    self.min_distance,
+                )
```

Comment:
I just added a "two_sided" flag to `nilearn.reporting.get_clusters_table`. Can/should that be leveraged here?

Reply:
Yes, that is perfect and I will use that, thanks!